confluent_kafka/avro/cached_schema_registry_client.py

```diff
@@ -59,32 +60,30 @@ def __init__(self, url, max_schemas_per_subject=1000):
         # subj => { schema => version }
         self.subject_to_schema_versions = defaultdict(dict)
 
-    def _send_request(self, url, method='GET', body=None, headers=None):
-        if body:
-            body = json.dumps(body)
-            body = body.encode('utf8')
-        _headers = dict()
-        _headers["Accept"] = ACCEPT_HDR
+        self._session = requests.Session()
+        with self._session as s:
+            if ca_location:
+                s.verify = ca_location
+            if (cert_location or key_location) is not None:
+                if (key_location and cert_location) is None:
```

Comment:
dito

Reply:
As with the `or` expression `and` does not evaluate to true or false here. Instead it will return the first argument to evaluated to false. If no false values can be found the last argument to evaluate true will be returned instead.

Reply:
Right, but you don't want to check key_location or cert_location for true/false, but for None.